core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -361,4 +364,32 @@ private static TableMetadata commit(TableOperations ops, UpdateTableRequest requ
 
     return ops.current();
   }
+
+  public static void commitTransaction(Catalog catalog, CommitTransactionRequest request) {
+    List<Transaction> transactions = Lists.newArrayList();
+
+    for (CommitTransactionRequest.CommitTableRequest tableChange : request.tableChanges()) {
+      Table table = catalog.loadTable(tableChange.identifier());
+      if (table instanceof BaseTable) {
+        UpdateTableRequest updateTableRequest =
+            new UpdateTableRequest(tableChange.requirements(), tableChange.updates());
+
+        Transaction transaction =
+            Transactions.newTransaction(
+                tableChange.identifier().toString(), ((BaseTable) table).operations());
+        transactions.add(transaction);
+
+        BaseTransaction.TransactionTable txTable =
+            (BaseTransaction.TransactionTable) transaction.table();
+
+        // this performs validations and makes temporary commits that are in-memory
+        commit(txTable.operations(), updateTableRequest);
+      } else {
+        throw new IllegalStateException("Cannot wrap catalog that does not produce BaseTable");
+      }
+    }
+
+    // only commit if validations passed previously
+    transactions.forEach(Transaction::commitTransaction);
```

Comment:
[doubt] let's say we had transactions {T1, T2, T3}  whose validations passed but then there was an update which made T2 conflict with current state, so when we actually commit T1 is applied but T2 can't now we need to rollback T1, how are we handling this case  ? 

Is there a write up for this ?

Reply:
@singhpk234 the intention here was to have a very simple approach that does only validate the requirements and nothing else. Hence we don't have a way of providing true atomicity here, which is left to the implementation details of a server. I've also added a comment to make this clear.